src/main/scala/net/snowflake/spark/snowflake/DefaultSource.scala

```diff
@@ -109,14 +109,17 @@ class DefaultSource(jdbcWrapper: JDBCWrapper)
       )
     }
 
-    def tableExists: Boolean = {
-      val conn = jdbcWrapper.getConnector(params)
-      try {
-        jdbcWrapper.tableExists(conn, table.toString)
-      } finally {
-        conn.close()
+    def tableExists: Boolean =
+      if (params.checkTableExistenceInCurrentSchemaOnly) {
+        DefaultJDBCWrapper.tableExistsInCurrentSchema(params, table.toString)
```

Comment:
Why DefaultJDBCWrapper ? This is unexpcted . Why not just jdbcWrapper. tableExistsInCurrentSchema

Reply:
In implementation, `jdbcWrapper` is `DefaultJDBCWrapper`. I will update to use `jdbcWrapper `